### PR TITLE
CSPL-2063: Update app framework doc to include azure blob usages

### PR DESCRIPTION
**Pull request by kumarajeet**
PR for covering updates for the following tasks:

https://splunk.atlassian.net/browse/CSPL-2063
**Update the app framework doc to include:**

- Examples of azure blob usage 
- Steps for using managed identity for azure blob

https://splunk.atlassian.net/browse/CSPL-2064
**Addition of new examples**
Added new examples similar to S3 to show usage of azure blob 
Removed smart store sections from azure blob examples

**Updated v3 to v4**

### CODE REVIEW

File: docs/AppFramework.md [hunk 1]
```diff
@@ -78,29 +127,34 @@ By default, the App Framework polls the remote object storage location for new o
...
    * The apps are split across 3 folders named `networkApps`, `clusterBase`, and `adminApps` . The apps placed into  `networkApps` and `clusterBase` are distributed to the cluster peers, but the apps in `adminApps` are for local use on the cluster manager instance only.
 
 4. Copy your Splunk app or add-on archive files to the App Source.
-   * In this example, the Splunk apps for the cluster peers are located at `bucket-app-framework-us-west-2/idxcAndCmApps/networkAppsLoc/`,  `bucket-app-framework-us-west-2/idxcAndCmApps/clusterBaseLoc/`, and the apps for the cluster manager are located at`bucket-app-framework-us-west-2/idxcAndCmApps/adminAppsLoc/`. They are all accessible through the end point `https://s3-us-west-2.amazonaws.com`.
+   * In this example, the Splunk apps for the cluster peers are located at `bucket-app-framework/idxcAndCmApps/networkAppsLoc/`,  `bucket-app-framework/idxcAndCmApps/clusterBaseLoc/`, and the apps for the cluster manager are located at`bucket-app-framework/idxcAndCmApps/adminAppsLoc/`. They are all accessible athrough the end point `https://s3-us-west-2.amazonaws.com` for s3 and https://mystorageaccount.blob.core.windows.net for azure blob.
```
Comment (sgontla) [inline docs/AppFramework.md:151]: nit `athrough` -> `through`

**Review by sgontla — APPROVED** [review]

File: docs/AppFramework.md (hunk 1 above)
Comment (kumarajeet) [reply docs/AppFramework.md]: Thanks @sgontla. Corrected.

**Review by vivekr-splunk — APPROVED** [review]

File: docs/AppFramework.md [hunk 2]
```diff
@@ -506,3 +630,84 @@ The App Framework does not preview, analyze, verify versions, or enable Splunk A
...
+
+Azure Managed identities can be used to provide IAM access to the blobs. With managed identities, the AKS nodes, that host the pods, can retrieve a OAuth token that provides authorization for the Splunk operator pod to read the app packages stored in the Azure Storage account. The key point here is that the AKS node is associated with a Managed Identity and this managed identity is given a `role` for read access called `Storage Blob Data Reader` to the azure storage account. 
+
+Here are the steps showing an example of assiging managed identity:
+
+*Assumptions:* 
+
+Familiarize yourself with [AKS managed identity concepts](https://learn.microsoft.com/en-us/azure/aks/use-managed-identity)
+
+The names used for resource group etc below are for examples purpose, please change them to the values as per your setup.
```
Comment (gaurav-splunk) [inline docs/AppFramework.md:643]: Maybe reword it to something like this - 
```
The names used below, such as the resource group, are for example purpose, please change them to the values as per your setup.
 ```

**Review by gaurav-splunk — APPROVED** [review]
